### PR TITLE
Make the sample azd compatible 

### PR DESCRIPTION
**Pull request by yiliuTo**
## Purpose

As subject, this pr is to make the sample compatible with azure developer cli.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Install the latest azd: https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/install-azd?tabs=winget-windows%2Cbrew-mac%2Cscript-linux&pivots=os-windows

```
mkdir spring-chatgpt-sample
cd spring-chatgpt-sample
azd auth login
azd config set alpha.springapp on
azd init --template https://github.com/yiliuTo/spring-chatgpt-sample.git -b support-azd
azd up
```

### CODE REVIEW

**Comment by yiliuTo** [comment]
There is a known issue for this PR that the app url displayed by AZD is wrong, which will be fixed by https://github.com/Azure/azure-dev/pull/2401

**Review by stliu — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -220,11 +220,92 @@ To run the demo, please follow these steps:
...
+java -jar spring-chatgpt-sample-cli/target/spring-chatgpt-sample-cli-0.0.1-SNAPSHOT.jar --from=/<path>/<to>/<your>/<documents> --to=doc_store.json
+```
+
+### Application Architecture
+
+This application utilizes the following Azure resources:
+
+- [**Azure Spring Apps**](https://docs.microsoft.com/azure/spring-apps/) to host the application
+- [**Azure OpenAI**](https://docs.microsoft.com/azure/cognitive-services/openai/) for ChatGPT
+- [**Azure Storage**](https://docs.microsoft.com/azure/storage/) for vector store
```
Comment (LGDoor) [inline README.md]: We need to clarify that the vector store is just a simple in-memory implementation. And the Azure Storage is used as a normal file volume for persistence. In fact the azure storage has no direct connection with the vector store.

File: README.md [hunk 2]
```diff
@@ -220,11 +220,92 @@ To run the demo, please follow these steps:
...
+
+Here's a high level architecture diagram that illustrates these components. Notice that these are all contained within a single [resource group](https://docs.microsoft.com/azure/azure-resource-manager/management/manage-resource-groups-portal), that will be created for you when you create the resources.
+
+!["Application architecture diagram"](assets/resources.png)
```
Comment (LGDoor) [inline README.md:290]: The same here. The `vector store` arrow may mislead the users about the purpose of the azure storage.

File: mvnw
Comment (LGDoor) [inline mvnw:1]: Is the maven wrapper mandatory?

Comment (yiliuTo) [reply mvnw:1]: @LGDoor yeah, we need it in azure.yaml to package the parent and common module, and to add maven wrapper in the sample, we can decrease a prerequisite about the maven installation.

**Review by LGDoor — APPROVED** [review]

**Pull request merged**